README.md

```diff
@@ -5,45 +5,164 @@
+https://user-images.githubusercontent.com/7043815/177197158-d3ea0736-629e-4dcb-bd5e-666993fbcfa2.mp4
 
-The framework’s unique software design provides many advantages, making it an ideal framework for deep-learning research and development in medical imaging:
 
-* **Rapid development** -
+### Examples - using FuseMedML-style components
 
-* **Flexible, customizable, and scalable** -
 
-* **Encourage sharing and collaboration** - 
+A multi head model FuseMedML style component, allows easy reuse across projects:
 
-* **Collection of off-the-shelf components and capabilities** - 
+```python
+ModelMultiHead(    
+    conv_inputs=(('data.input.img', 1),),                                       # input to the backbone model   
+    backbone=BackboneResnet3D(in_channels=1),                                   # PyTorch nn Module    
+    heads=[                                                                     # list of heads - gives the option to support multi task / multi head approach
+               Head3DClassifier(head_name='classification',                     
+                                conv_inputs=[("model.backbone_features", 512)]  # Input to the classification head
+                                ,),                                              
+          ]
+)
+```
 
-* **Standardized evaluation** - 
+Our default loss implementation - creates an easy wrap around a callable function, while being FuseMedML style
+```python
+LossDefault(
+    pred='model.logits.classification',          # input - model prediction scores
+    target='data.label',                         # input - ground truth labels
+    callable=torch.nn.functional.cross_entropy   # callable - function that will get the prediction scores and labels extracted from batch_dict and compute the loss
+)
+```
 
-* **Medical imaging expertise** - 
+An example metric that can be used
+```python
+MetricAUCROC(
+    pred='model.output', # input - model prediction scores
+    target='data.label'  # input - ground truth labels
+)
+```
 
-* **Compatibility with alternative frameworks**
+Note that several components return answers directly and not write it into the nested dictionary. This is perfectly fine, and to allow maximum flexibility we do not require any usage of output path keys.
 
-# Citation
-If you use FuseMedML in scientific context, please consider citing us:
-```bibtex
-@misc{https://doi.org/10.5281/zenodo.5146491,
-  doi = {10.5281/ZENODO.5146491},
-  url = {https://zenodo.org/record/5146491},
-  author = {IBM Research,  Haifa},
-  title = {FuseMedML: https://github.com/IBM/fuse-med-ml},
-  publisher = {Zenodo},
-  year = {2021},
-  copyright = {Apache License 2.0}
-}
+* metrics
+* loss
+* multihead model
+
+### Creating a custom FuseMedML component
+
+Creating custom FuseMedML components is easy - in the following example we add a new data processing operator:
+
+
+A data pipeline operator
+```python
```

Comment:
I like this! Well motivated and explicit. Only problem is I don't know how it fits in with the other pieces. I assume there's some kind of pipeline object that this can be added as a list in. Can that be shown also?